test/export/StructureDefinitionExporter.test.ts

```diff
@@ -759,6 +769,75 @@ describe('StructureDefinitionExporter', () => {
     );
   });
 
+  it('should log a debug message when we detect a circular dependency in OnlyRules that might result in incomplete definitions', () => {
+    const profile1 = new Profile('FooQuantity');
+    profile1.parent = 'Quantity';
+    const p1ContainsRule = new ContainsRule('extension');
+    p1ContainsRule.items.push('QuantityExtension');
+    const p1OnlyRule = new OnlyRule('extension[QuantityExtension].valueQuantity');
+    p1OnlyRule.types = [{ type: 'BarQuantity' }];
+    const p1FixedValueRule = new FixedValueRule('extension[QuantityExtension].valueQuantity.code');
+    p1FixedValueRule.fixedValue = new FshCode('mg');
+    profile1.rules = [p1ContainsRule, p1OnlyRule, p1FixedValueRule];
+    doc.profiles.set(profile1.name, profile1);
+
+    const profile2 = new Profile('BarQuantity');
+    profile2.parent = 'Quantity';
+    const p2ContainsRule = new ContainsRule('extension');
+    p2ContainsRule.items.push('QuantityExtension');
+    const p2OnlyRule = new OnlyRule('extension[QuantityExtension].valueQuantity');
+    p2OnlyRule.types = [{ type: 'FooQuantity' }];
+    const p2FixedValueRule = new FixedValueRule('extension[QuantityExtension].valueQuantity.code');
+    p2FixedValueRule.fixedValue = new FshCode('mg');
+    profile2.rules = [p2ContainsRule, p2OnlyRule, p2FixedValueRule];
+    doc.profiles.set(profile2.name, profile2);
+
+    const extension = new Extension('QuantityExtension');
+    const extOnlyRule = new OnlyRule('value[x]');
+    extOnlyRule.types = [{ type: 'Quantity' }];
+    extension.rules = [extOnlyRule];
+    doc.extensions.set(extension.name, extension);
+
+    withDebugLogging(() => exporter.export());
+
+    const lastLog = loggerSpy.getLastLog();
+    expect(lastLog.level).toMatch(/debug/);
+    expect(lastLog.message).toMatch(/Warning: Circular .* BarQuantity and FooQuantity/);
+
+    expect(loggerSpy.getLastMessage()).toMatch(/Warning: Circular .* BarQuantity and FooQuantity/);
```

Comment:
I would have never thought to just abbreviate a test message in a regex like that. Now I know.